### PR TITLE
dotCMS/core#16076 Change parseContainer directive to use path instead of id in page layout for Containers as files

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts [hunk 1]
```diff
@@ -105,7 +109,10 @@ export class DotEditLayoutService {
                             .map(
                                 (dotContainersColumnBox: DotContainerColumnBox) =>
                                     <DotPageContainer>{
-                                        identifier: dotContainersColumnBox.container.identifier,
+                                        identifier:
+                                            dotContainersColumnBox.container.source === 'FILE'
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts]: source should be a enum

File: src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts [hunk 2]
```diff
@@ -105,7 +109,10 @@ export class DotEditLayoutService {
...
+                                                ? dotContainersColumnBox.container.path
+                                                : dotContainersColumnBox.container.identifier,
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts]: maybe we should create a methos for it something like:

```
getContainerId(dotContainersColumnBox: DotContainer): string {
    return dotContainersColumnBox.container.source === 'FILE' ? dotContainersColumnBox.container.path : 
         dotContainersColumnBox.container.identifier;
}
```

File: src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts [hunk 3]
```diff
@@ -105,7 +109,10 @@ export class DotEditLayoutService {
                             .map(
                                 (dotContainersColumnBox: DotContainerColumnBox) =>
                                     <DotPageContainer>{
-                                        identifier: dotContainersColumnBox.container.identifier,
+                                        identifier:
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts]: I am not reaaly sure if all this change is test into the spec file, I not see any expect for any path for example.

File: src/app/portlets/dot-edit-page/template-containers-cache.service.ts [hunk 4]
```diff
@@ -9,7 +9,15 @@ export class TemplateContainersCacheService {
     private containers: { [key: string]: { container: DotContainer } };
 
     set(containers: { [key: string]: { container: DotContainer } }): void {
-        this.containers = containers;
+        const mappedContainers: { [key: string]: { container: DotContainer } } = {};
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/template-containers-cache.service.ts]: should this be a interface?

File: src/app/portlets/dot-edit-page/template-containers-cache.service.ts [hunk 5]
```diff
@@ -9,7 +9,15 @@ export class TemplateContainersCacheService {
...
+        Object.keys(containers).forEach(function(item: string) {
+            if (containers[item].container.source === 'FILE') {
+                mappedContainers[containers[item].container.path] = containers[item];
+            } else {
+                mappedContainers[item] = containers[item];
+            }
+        });
+        this.containers = mappedContainers;
```
Comment (freddyucv) [inline src/app/portlets/dot-edit-page/template-containers-cache.service.ts]: why not assing directly to this.containers?
if we create the getContainerId method that maybe I said before we can use it here too

```
mappedContainers[this.getContainerId(containers[item])] = containers[item];
```

File: src/app/portlets/dot-site-browser/directives/dot-dnd-files-folders.directive.spec.ts [hunk 6]
```diff
@@ -1,8 +1,7 @@
...
 
 describe('DotDndFilesFoldersDirective', () => {
-    it('should create an instance', () => {
-        const directive = new DotDndFilesFoldersDirective();
-        expect(directive).toBeTruthy();
-    });
+    // it('should create an instance', () => {
+    //     const directive = new DotDndFilesFoldersDirective();
+    //     expect(directive).toBeTruthy();
+    // });
```
Comment (freddyucv) [inline src/app/portlets/dot-site-browser/directives/dot-dnd-files-folders.directive.spec.ts]: remove commented code

File: src/app/shared/models/container/dot-container.model.ts [hunk 7]
```diff
@@ -2,7 +2,9 @@ export interface DotContainer {
     categoryId?: string;
     deleted?: boolean;
     friendlyName?: string;
+    path?: string;
     identifier: string;
     name: string;
     type: string;
+    source: string;
```
Comment (freddyucv) [inline src/app/shared/models/container/dot-container.model.ts]: should be a enum

File: src/app/view/components/dot-container-selector/dot-container-selector.component.ts [hunk 8]
```diff
@@ -83,15 +79,23 @@ export class DotContainerSelectorComponent implements OnInit {
...
+            this.currentContainers = this.setIdentifierReference(items.splice(0));
             this.totalRecords = this.totalRecords || this.paginationService.totalRecords;
         });
     }
+
+    private setIdentifierReference(items: DotContainer[]): any {
+        return items.map(dotContainer => {
+            dotContainer.identifier =
+                dotContainer.source === 'FILE' ? dotContainer.path : dotContainer.identifier;
+            return dotContainer;
```
Comment (freddyucv) [inline src/app/view/components/dot-container-selector/dot-container-selector.component.ts]: we should use the getContainerId method here too

File: src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts (hunk 3 above)
Comment (hmoreras) [reply src/app/portlets/dot-edit-page/shared/services/dot-edit-layout.service.ts]: this, https://github.com/dotCMS/core-web/pull/938/files#diff-9fdb7f7a5d8c1c883513d66fe700c574R244, evaluate the change

File: src/app/portlets/dot-edit-page/template-containers-cache.service.ts (hunk 5 above)
Comment (hmoreras) [reply src/app/portlets/dot-edit-page/template-containers-cache.service.ts]: Object.keys creates a new scope, `this.containers` is private

**Pull request merged**